src/controllers/dynakube/dynakube_controller.go

```diff
@@ -301,6 +276,37 @@ func (controller *DynakubeController) reconcileDynaKube(ctx context.Context, dkS
 	dkState.Update(upd, "dynakube phase changed")
 }
 
+func (controller *DynakubeController) reconcileOneAgent(ctx context.Context, dkState *status.DynakubeState) (err error) {
+	if dkState.Instance.HostMonitoringMode() {
+		upd, err := oneagent.NewOneAgentReconciler(
+			controller.client, controller.apiReader, controller.scheme, dkState.Instance, daemonset.DeploymentTypeHostMonitoring,
+		).Reconcile(ctx, dkState)
+		if dkState.Error(err) {
+			return nil
+		}
+		dkState.Update(upd, "host monitoring reconciled")
+	} else if dkState.Instance.CloudNativeFullstackMode() {
+		upd, err := oneagent.NewOneAgentReconciler(
+			controller.client, controller.apiReader, controller.scheme, dkState.Instance, daemonset.DeploymentTypeCloudNative,
+		).Reconcile(ctx, dkState)
+		if dkState.Error(err) {
+			return nil
+		}
+		dkState.Update(upd, "cloud native fullstack monitoring reconciled")
```

Comment:
the bodies of these branches could be moved into their own functions to make them more readable. Additionally, the only things that really changes, as far as I can see, is the deployment type and the log line, so there could be potential to further abstract these lines

Reply:
Out of scope of this PR.